x-pack/plugins/index_management/public/application/sections/home/template_list/template_details/tabs/tab_summary.tsx

```diff
@@ -21,84 +25,184 @@ interface Props {
+            <>
+              <EuiDescriptionListTitle data-test-subj="componentsTitle">
+                <FormattedMessage
+                  id="xpack.idxMgmt.templateDetails.summaryTab.componentsDescriptionListTitle"
+                  defaultMessage="Components"
+                />
+              </EuiDescriptionListTitle>
+              <EuiDescriptionListDescription>
+                {composedOf && composedOf.length > 0 ? (
+                  <ul>
+                    {composedOf.map((component) => (
+                      <li key={component}>
+                        <EuiTitle size="xs">
+                          <span>{component}</span>
+                        </EuiTitle>
+                      </li>
+                    ))}
+                  </ul>
+                ) : (
+                  i18nTexts.none
+                )}
+              </EuiDescriptionListDescription>
+            </>
+          )}
+        </EuiDescriptionList>
+      </EuiFlexItem>
+
+      <EuiFlexItem>
+        <EuiDescriptionList textStyle="reverse">
+          {/* ILM Policy (only for legacy as composable template could have ILM policy
+              inside one of their components) */}
+          {isLegacy && (
+            <>
+              <EuiDescriptionListTitle>
+                <FormattedMessage
+                  id="xpack.idxMgmt.templateDetails.summaryTab.ilmPolicyDescriptionListTitle"
+                  defaultMessage="ILM policy"
+                />
+              </EuiDescriptionListTitle>
+              <EuiDescriptionListDescription>
+                {ilmPolicy && ilmPolicy.name ? (
+                  <EuiLink href={getILMPolicyPath(ilmPolicy.name)}>{ilmPolicy.name}</EuiLink>
+                ) : (
+                  i18nTexts.none
+                )}
+              </EuiDescriptionListDescription>
+            </>
+          )}
 
-      {/* // ILM Policy */}
-      <EuiDescriptionListTitle>
-        <FormattedMessage
-          id="xpack.idxMgmt.templateDetails.summaryTab.ilmPolicyDescriptionListTitle"
-          defaultMessage="ILM policy"
-        />
-      </EuiDescriptionListTitle>
-      <EuiDescriptionListDescription>
-        {ilmPolicy && ilmPolicy.name ? (
-          <EuiLink href={getILMPolicyPath(ilmPolicy.name)}>{ilmPolicy.name}</EuiLink>
-        ) : (
-          <NoneDescriptionText />
-        )}
-      </EuiDescriptionListDescription>
+          {/* Has data stream? (only for composable template) */}
+          {isLegacy !== true && (
+            <>
+              <EuiDescriptionListTitle>
+                <FormattedMessage
+                  id="xpack.idxMgmt.templateDetails.summaryTab.dataStreamDescriptionListTitle"
+                  defaultMessage="Data stream"
+                />
+              </EuiDescriptionListTitle>
+              <EuiDescriptionListDescription>
+                {hasDatastream ? i18nTexts.yes : i18nTexts.no}
+              </EuiDescriptionListDescription>
+            </>
+          )}
```

Comment:
Not a copy edit, but it would be great to see if an index template is in use for any matching data streams and list those data streams here.

Reply:
That'd be a great enhancement we could add, not sure we could squeeze it for `7.9` though. I thought we didn't have that information yet available on ES side. I'll check it out 👍